src/embedded_flash/buffer.rs

```diff
@@ -13,6 +13,10 @@
 // limitations under the License.
 
 use super::{Index, Storage, StorageError, StorageResult};
+#[cfg(not(feature = "std"))]
+use alloc::boxed::Box;
+#[cfg(not(feature = "std"))]
+use core as std;
```

Comment:
I find this confusing. The code should rather use `core`, and making sure that `core` is recognized in a std context for the tests.

Reply:
Used `core`. I don't know why I thought I should use `std` with `features = "std"`.